packages/cli/src/databases/entities/WorkflowEntity.ts

```diff
@@ -84,6 +88,30 @@ export class WorkflowEntity extends AbstractEntity implements IWorkflowDb {
 		transformer: sqlite.jsonColumn,
 	})
 	pinData: ISimplifiedPinData;
+
+	/**
+	 * Hash of editable workflow state.
+	 */
+	hash: string;
+
+	@AfterLoad()
+	@AfterUpdate()
+	@AfterInsert()
+	setHash(): void {
+		const { name, active, nodes, connections, settings, staticData, pinData } = this;
+
+		const state = JSON.stringify({
+			name,
+			active,
+			nodes,
+			connections,
+			settings,
+			staticData,
+			pinData,
+		});
```

Comment:
I think we should simplify this to use only nodes and connections. Maybe pinData.

This reduces the chance of having issues with unnecessary warnings such as name changes or activation status.

Reply:
Thanks! As discussed, any editable parts of the workflow that we exclude from the hash will cause changes to those parts to be overwritable, which we likely do not want, so keeping it strict for now.